server/src/middleware/http/proxy_get_request.rs

```diff
@@ -53,29 +55,40 @@ pub struct InvalidPath(String);
 /// See [`ProxyGetRequest`] for more details.
 #[derive(Debug, Clone)]
 pub struct ProxyGetRequestLayer {
-	path: String,
-	method: String,
+	// path => method mapping
+	methods: Arc<FxHashMap<String, String>>,
 }
 
 impl ProxyGetRequestLayer {
 	/// Creates a new [`ProxyGetRequestLayer`].
 	///
-	/// See [`ProxyGetRequest`] for more details.
-	pub fn new(path: impl Into<String>, method: impl Into<String>) -> Result<Self, InvalidPath> {
-		let path = path.into();
-		if !path.starts_with('/') {
-			return Err(InvalidPath(path));
+	/// The request `GET /path` is redirected to the provided method.
+	/// Fails if the path does not start with `/`.
+	pub fn new<P, M>(methods: impl IntoIterator<Item = (P, M)>) -> Result<Self, InvalidPath>
+	where
+		P: Into<String>,
+		M: Into<String>,
+	{
+		let methods = methods
```

Comment:
I believe we can check for path duplicates. Currently, we'll keep in the hashmap just the last entry over which we iterate.

For example, we currently accept:

```rust
[("/health", "system_health"), ("/health", "system2")]
```

This is a misconfig from the user perspective, atm the implementation keeps the `/health` -> `system2` instead of returning an error

Reply:
We could also just use a `Vec` here because we don't expect that many endpoints exposed anyway. 
HashMap is probably a bit overkill but either will work...

Reply:
Added check and error type for duplicate path